.github/workflows/main.yml

```diff
@@ -0,0 +1,93 @@
+name: Build Upstream
+run-name: Running ${{ github.workflow }} by ${{ github.actor }}
+on: [push]
+jobs:
+  Build:
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "Job triggered by ${{ github.event_name }} event, running on ${{ runner.os }}"
+      - run: echo "The name of your branch is ${{ github.ref }} and your repository is ${{ github.repository }}"
+      - name: Setup environment
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y make build-essential python3 python3-pyelftools yamllint
+      - name: Setup Cross compilers
+        run: |
+          pushd /tmp
+          curl -Lo gcc-arm-none-linux-gnueabihf.tar.xz "https://developer.arm.com/-/media/Files/downloads/gnu/12.2.rel1/binrel/arm-gnu-toolchain-12.2.rel1-x86_64-arm-none-linux-gnueabihf.tar.xz"
```

Comment:
Seems like a lot to download each run from poor arm.com.

Also these will go out of date over time. Why not use the latest cross-compilers that come with the latest Ubuntu?

`apt install gcc-aarch64-linux-gnu gcc-arm-none-eabi`

Reply:
If you think this will be a problem, I can investigate caching.  I use it with the subproject repos as it is easy to use with git.  See line 26.  It will be a little more complicated to use for curl, but I did have that on my list to try and forgot about it.

On which toolchain, I wanted to follow the recommend build instructions we have in our readme.md.  If we want to switch to debian packaged cross compilers, let's make that a policy decision and also update the readme.md.  If so, I think that should be a separate PR.

Reply:
For the toolchains, we might still want the readme.md to include links to a specific known working version, just in case someone is using some old OS. For the github actions we know it will be a modern Ubuntu image with up to date compilers already available.

Reply:
If we switch to using debian packaged cross compiler, then we also need to override the default CROSS_COMPILE_** variables, right?  It doesn't work for me if I just install native debian instead of arm.com.

Reply:
Yeah, the compiler shipped with Debian has a little different prefix than the default names we use in the Makefile.